File: src/types/capture.ts
export interface Capture {
  id: string
  sessionId: string
  screenshotPath: string
  monitorName: string
  monitorX: number
  monitorY: number
  monitorWidth: number
  monitorHeight: number
  scaleFactor: number
  windowTitle: string | null
  createdAt: string
}

export interface MonitorInfo {
  x: number
  y: number
  width: number
  height: number
  scaleFactor: number
  name: string
}

File: src-tauri/src/models/capture.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Capture {
    pub id: String,
    pub session_id: String,
    pub screenshot_path: String,
    pub monitor_name: String,
    pub monitor_x: i32,
    pub monitor_y: i32,
    pub monitor_width: i32,
    pub monitor_height: i32,
    pub scale_factor: f64,
    pub window_title: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MonitorInfo {
    pub x: i32,
    pub y: i32,
    pub width: i32,
    pub height: i32,
    pub scale_factor: f64,
    pub name: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateCapturePayload {
    pub session_id: String,
    pub screenshot_path: String,
    pub monitor_name: String,
    pub monitor_x: i32,
    pub monitor_y: i32,
    pub monitor_width: i32,
    pub monitor_height: i32,
    pub scale_factor: f64,
    pub window_title: Option<String>,
}
